services/app-api/utils/auth/authorization.ts

```diff
@@ -81,8 +84,14 @@ export const hasPermissions = (
       mcrUserRole = UserRoles.STATE_USER;
     }
 
-    if (allowedRoles.includes(mcrUserRole)) {
-      isAllowed = true;
+    if (state) {
```

Comment:
Idea: You can de-nest the logic even further by rejecting early for each failure case:

```typescript
export const hasPermissions = (
  event: APIGatewayProxyEvent,
  allowedRoles: UserRoles[],
  state?: string
) => {
  const apiKey = event?.headers?.["x-api-key"];
  if (!apiKey) {
    return false;
  }

  const decoded = jwt_decode(apiKey) as DecodedToken;
  const idmUserRoles = decoded["custom:cms_roles"];
  const idmUserState = decoded["custom:cms_state"];
  let mcrUserRole = idmUserRoles
    ?.split(",")
    .find((role) => role.includes("mdctmcr")) as UserRoles;

  // consolidate "STATE_REP" role into "STATE_USER" role
  if (mcrUserRole === UserRoles.STATE_REP) {
    mcrUserRole = UserRoles.STATE_USER;
  }

  if (!allowedRoles.includes(mcrUserRole)) {
    return false;
  }

  if (state && !idmUserState?.includes(state)) {
    return false;
  }

  return true;
};
```

Reply:
I'm a little wary of using checks like these (where true is default) because then if we miss some type of logic, or we don't update this in time when there is another attribute to check for, then it defaults to true instead of false.